unified-runtime/source/adapters/level_zero/v2/queue_immediate_in_order.cpp

```diff
@@ -1036,6 +1037,77 @@ ur_result_t ur_queue_immediate_in_order_t::enqueueCommandBufferExp(
   return UR_RESULT_SUCCESS;
 }
 
+ur_result_t ur_queue_immediate_in_order_t::enqueueKernelLaunchWithArgsExp(
```

Comment:
I think it would be better to move most of the implementation to command_manager.cpp. If we do that, we can:
1. avoid taking kernel->mutex twice (commandListLocked->appendKernelLaunch does that internally)
2. avoid creating pending_allocations vector - this was only needed for urKernelSetArgMemObj where we need to delay calling zeKernelSetArgumentValue until actual kernel submission. Now, we can just call it directly.

I'm okay doing this myself, though, after this PR is merged. Just wanted to note it somewhere.